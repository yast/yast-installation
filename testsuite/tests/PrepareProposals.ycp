{

// testedfiles: AddOnProduct.ycp

include "testsuite.ycp";

map READ = $[
    "target" : $[
	"size" : 0,
    ],
];

map WRITE = $[];

map EXEC = $[
    "target" : $[
	"bash_output" : $["exit" : 0, "stdout" : "", "stderr" : ""],
    ],
];

TESTSUITE_INIT([READ, WRITE, EXEC], nil);

import "AddOnProduct";

list<map> proposals = [
  $[
    "mode":"installation,demo,autoinstallation",
    "name":"initial",
    "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"],
    "proposal_tabs":[
      $["label":"Overview", "proposal_modules":["partitions", "software_simple", "language_simple"]],
      $["label":"Expert", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"]]
    ],
    "stage":"initial"
  ],
  $[
    "name":"network",
    "proposal_modules":[
      $["name":"lan", "presentation_order":"20"],
      $["name":"dsl", "presentation_order":"30"],
      $["name":"isdn", "presentation_order":"40"],
      $["name":"modem", "presentation_order":"50"],
      $["name":"remote", "presentation_order":"60"],
      $["name":"firewall", "presentation_order":"10"],
      $["name":"proxy", "presentation_order":"70"]
    ],
    "stage":"continue,normal"
  ],
];

TEST(``(AddOnProduct::PrepareProposals(proposals)), [], nil);

}
