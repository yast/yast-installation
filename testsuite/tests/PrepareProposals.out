Return	[$["archs":"", "mode":"installation", "name":"initial", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"], "proposal_tabs":[$["label":"Overview", "proposal_modules":["partitions", "software_simple", "language_simple"]], $["label":"Expert", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"]]], "stage":"initial"], $["archs":"", "mode":"demo", "name":"initial", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"], "proposal_tabs":[$["label":"Overview", "proposal_modules":["partitions", "software_simple", "language_simple"]], $["label":"Expert", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"]]], "stage":"initial"], $["archs":"", "mode":"autoinstallation", "name":"initial", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"], "proposal_tabs":[$["label":"Overview", "proposal_modules":["partitions", "software_simple", "language_simple"]], $["label":"Expert", "proposal_modules":["hwinfo", "keyboard", "mouse", "partitions", "software", "bootloader", "timezone", "language", "runlevel"]]], "stage":"initial"], $["archs":"", "mode":"", "name":"network", "proposal_modules":[$["name":"lan", "presentation_order":"20"], $["name":"dsl", "presentation_order":"30"], $["name":"isdn", "presentation_order":"40"], $["name":"modem", "presentation_order":"50"], $["name":"remote", "presentation_order":"60"], $["name":"firewall", "presentation_order":"10"], $["name":"proxy", "presentation_order":"70"]], "stage":"continue"], $["archs":"", "mode":"", "name":"network", "proposal_modules":[$["name":"lan", "presentation_order":"20"], $["name":"dsl", "presentation_order":"30"], $["name":"isdn", "presentation_order":"40"], $["name":"modem", "presentation_order":"50"], $["name":"remote", "presentation_order":"60"], $["name":"firewall", "presentation_order":"10"], $["name":"proxy", "presentation_order":"70"]], "stage":"normal"]]
